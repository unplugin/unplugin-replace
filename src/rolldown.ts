/**
 * This entry file is for Rolldown plugin.
 *
 * @module
 */

import unplugin from './index'

/**
 * Rolldown plugin
 *
 * @example
 * ```ts
 * // rolldown.config.js
 * import Replace from 'unplugin-replace/rolldown'
 *
 * export default {
 *   plugins: [Replace()],
 * }
 * ```
 */
export default unplugin.rolldown
