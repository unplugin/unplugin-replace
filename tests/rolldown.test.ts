import path from 'node:path'
import { expect, test } from 'vitest'
import { rolldown } from 'rolldown'
import UnpluginReplace from '../src/rolldown'

test('rolldown', async () => {
  const build = await rolldown({
    input: path.resolve(__dirname, 'fixtures/main.js'),
    external: ['node:process'],
    plugins: [
      UnpluginReplace({
        'process.platform': '"darwin"',
      }),
    ],
  })
  const { output } = await build.generate({ format: 'es' })
  const code = output[0].code
  expect(code).toMatchSnapshot()
  expect(code).not.contains('process.platform')
  expect(code).contains('"darwin"')
})
