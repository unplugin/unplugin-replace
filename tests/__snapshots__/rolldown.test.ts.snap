// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`rolldown 1`] = `
"import { default as process } from "node:process";

// tests/fixtures/main.js
const platform = 'darwin';
console.log('hello "darwin"', platform);
console.log(DEV);

export { platform };"
`;
